kendzi3d-plugin/src/main/java/kendzi/josm/kendzi3d/ui/layer/CameraLayer.java

```diff
@@ -256,13 +256,9 @@ public void layerAdded(LayerAddEvent evt) {
      */
     @Override
     public void layerRemoving(LayerRemoveEvent evt) {
-        if (evt.getRemovedLayer() instanceof OsmDataLayer && !isOsmDataLayer()) {

```

Comment:
checking if is last osm layer had some sense here. But don't change it i will make pach later.

Reply:
The problem with this was, that JOSM had thrown an exception, so it did not work.  (may be it had worked with an earlier version of JOSM this way, but it does not so anymore..).

I've pasted the exception thrown by JOSM in response to executing this to the commit message, so you may find another fix to the bug that works just as well.

Reply:
This happened with the previous code

ReportedException [thread=Thread[AWT-EventQueue-1,6,main], exception=java.lang.IllegalArgumentException: CameraLayer [name=Kendzi3d camera layer, associatedFile=null] is not managed by us., methodWarningFrom=null]
        at org.openstreetmap.josm.tools.bugreport.BugReport.intercept(BugReport.java:173)
        at org.openstreetmap.josm.gui.layer.LayerManager.fireLayerRemoving(LayerManager.java:477)
        at org.openstreetmap.josm.gui.layer.LayerManager.realRemoveSingleLayer(LayerManager.java:277)
....

Full stack trace [here](https://github.com/cm8/kendzi3d/commit/7ba362f21375b70768862beaea0211bc032cd41a)

Reply:
#75 is included in my last commit,  with isOsmDataLayer() method name unchanged..